adanet-co2-monitor.ino

```diff
@@ -104,6 +115,61 @@ void printfAligned(const uint8_t size, const Alignment alignment, const int16_t
   display.print(buffer);
 }
 
+void co2HistoryAdd(const uint16_t co2)
+{
+  co2HistoryFifo[co2HistoryHead] = co2;
+  ++co2HistoryHead %= UPDATES_PER_WEEK;
+
+  if (++co2HistorySize > UPDATES_PER_WEEK) 
+  {
+    co2HistorySize = UPDATES_PER_WEEK;
+  }
+}
+
+uint16_t co2HistoryRead(const uint16_t index) 
+{
+  int16_t idx = co2HistoryHead - 1 - index;
+
+  if (idx < 0) 
```

Comment:
Should this be a `while` (in case `index > UPDATES_PER_WEEK` for some reason)?

Reply:
updated